### DISCUSSION TITLE
Duplicated/overriden API routes have wrong values in JSON schema because of incorrect order

### DISCUSSION THREAD

**Discussion by alexkreidler** [fastapi/fastapi#9606] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class Author(BaseModel):
    id: int
    name: str

@app.get("/author", response_model=Author)
def whats_up():
    return Author(id=1, name="Test")

@app.get("/author", response_model=int)
def test_2():
    return 5
```

### Description

When an app or router has duplicate routes, FastAPI calls and serves the first route function registered in response to requests. However, it publishes an OpenAPI schema with the return data type of the last route registered. This will introduce bugs into any generated clients, and is very problematic for people who would like to programmatically create some default routes, like CRUD actions on a database using SQLModel, and override some of them to add additional logic.

1. Open the OpenAPI docs at `http://localhost:8000/docs`
2. See that the response example value for the `/author` endpoint is an integer, and the schema is an integer
3. That value should be an `Author` object, not an integer, to match the actual behavior of FastAPI (next step)
4. Click "Try it out" and "Execute" and see that the response is this JSON object:
```json
{
  "id": 1,
  "name": "Test"
}
```

### Operating System

Linux

### FastAPI Version

0.95.2

### Python Version

Python 3.11.3

**Comment by JavierSanchezCastro** [comment]
Yes, it seems to be a little bug
